### DISCUSSION TITLE
그룹#2 회고

### DISCUSSION THREAD

**Discussion by godrm** [codesquad-members-2022/swift-pokergame#82] [Show and tell]
여기에 각자 이번주 회고를 기록해주세요.
- 학습하면서 잘했던 점, 좋았던 점은 무엇인가요? (다음주에도 계속 유지해보세요)
- 학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 - 그래서 어떻게 개선할 수 있을까요? (다음주에는 개선해봐요)
- 새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? (다음주에 시도해봐요)
- 학습하는 데 조금이라도 도움이 된 사람이 있다면 칭찬해주세요!

**Comment by shingha1124** [comment]
- **학습하면서 잘했던 점, 좋았던 점은 무엇인가요? (다음주에도 계속 유지해보세요)**
  - 이미 개발 경험이 있어 습관이 들어버린 부분을 조금 씩 걷어내고 있는게 보이는것 같다
  - 조금더 깔끔한 코드를 배워나가는거같아서 좋았다
  - 처음 접해보는 프로토콜들을 알게 되었고, 사용해보면서 어디에 사용하면 좋을지 생각 해볼수 있었다
  - 하드코딩하는 부분을 거진 다 제거하고, 타입으로 구분하여 처리하였다
- **학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 - 그래서 어떻게 개선할 수 있을까요? (다음주에는 개선해봐요)**
  - 익숙한것만 쓰지말고, 다른방법, 다른코드를 많이 활용해서 적절한곳에 사용할 수 있도록 노력해야겠다
  - 함수명이나 변수명등의 이름을 딱봐도 뭐하는 것인지 알 수 있도록 이름짓는 연습을 해야할 것 같다
- **새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? (다음주에 시도해봐요)**
  - XCTest를 잘 활용해 봐야겠다
    - 사용할수록 뭔가 편해지는게 느껴지기는 하다. 
    - 현재는 작은 프로젝트지만 추후 큰 프로젝트로 가게 될 경우에 편리하게 사용할 수 있을것 같다
    - 활용법을 익혀두자
- **학습하는 데 조금이라도 도움이 된 사람이 있다면 칭찬해주세요!**
  - 다른분들 PR을 보며 여러가지 코딩방법을 작성하지않고?? 확인해 볼수 있었다
  - 질문을 받고 답하는 과정에서도 많이 복습하며 성장할수 있었습니다

**Reply by Jinsujin** [reply]
싱하 항상 감사합니다! PR 도 흘끔흘끔 보면서 다 이해는 아직 못했지만 참고하고 있어요 👍

**Comment by yeaju-Choi** [comment]
### 학습하면서 잘했던 점, 좋았던 점은 무엇인가요? (다음주에도 계속 유지해보세요)
- 그동안 무지성 코딩을 했는데, 어떤 관점에서 구현해야하고, 어떤 흐름대로 해야하는지 배워서 좋았다. 
- 토끼책을 샀다. 
    
### 학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 - 그래서 어떻게 개선할 수 있을까요? (다음주에는 개선해봐요)
- 체력 관리를 잘 해야될것 같다. 다음주부터 아침 운동을 하기로 했다.  
- 학습했던것을 적용하면서 짜고싶은 욕심이 생기다보니 구현을 하기 전에 시간이 너무 오래걸렸다. 
    
### 새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? (다음주에 시도해봐요)
- XCTest 처음 접해보는데, 아직 익숙하지 않아 더 연습해봐야 겠다. 
- 테스트코드를 먼저 작성하고 그에 필요한 메소드를 구현해보기 
     
### 학습하는 데 조금이라도 도움이 된 사람이 있다면 칭찬해주세요!
- 수업을 듣지못한 저를 위해 새벽까지 객체지향 강의해주신 로사 정말정말 감사합니다❤️

**Reply by Jinsujin** [reply]
체즈 저도 도움이 되었어요! 말로 내뱉어 보니 확실히 정리가 더 잘되더라구요, 다음에도 같이 공부해요! 🙌🏼

**Comment by Jinsujin** [comment]
### 학습하면서 잘했던 점, 좋았던 점은 무엇인가요? (다음주에도 계속 유지해보세요)
- 객체지향에 대해 많은 고민을 하고 코드에 적용시키려고 노력해 보았다.
- TDD 를 처음 적용해 보았다.
  -  구체적인 결과를 먼저 예상하고 코드를 구현해 보니, 조금 더 고민의 시간이 줄어들었다.
- 아침에 하루에 대한 계획을 짜고 시작해보니 좀 더 계획적인 하루를 보냈어서 좋았다.

### 학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 - 그래서 어떻게 개선할 수 있을까요? (다음주에는 개선해봐요)
- 밤 늦게까지 코딩을 해서 다음날까지 컨디션이 안좋은 때가 있었다. 일찍 자보도록 해야겠다

### 새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? (다음주에 시도해봐요)
- XCTest 를 많이 활용하지 못했다
  - 여러 테스트를 해보아야 겠다(BICEP)

### 학습하는 데 조금이라도 도움이 된 사람이 있다면 칭찬해주세요!
- 체즈에게 객체지향에 대해 알려주면서, 그동안 머리속으로 생각한 개념을 정리할 수 있었다.
- 피그백: 몸이 좋지 않았지만 빠지지 않고 꾸준히 참여해 줘서 학습 환경 조성에 도움을 주었다.
- 싱하: 항상 늦게까지 같이 공부해주셔서 든든했고 질문에 친철히 대답해 주셨다 😊
- 메이스: 라이브 코딩 방송! 좋아요 구독

**Comment by piggyse** [comment]
### 학습하면서 잘했던 점, 좋았던 점은 무엇인가요? 
- XCTest , 성능테스트와같이 무서운? 단어들이 나와도 겁먹지 않고 시도했다. 
- 목표를 과하게 잡지 않아서 나의 멘탈에 좋았다.

### 학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 - 그래서 어떻게 개선할 수 있을까요? (다음주에는 개선해봐요)

- 한번 집중이 무너지면 공부하지도, 쉬지도 않는 애매한 상태로 시간을 버린 것.
	-  다음주에는 쉴땐 쉬고 놀땐 놀자!

### 새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? (다음주에 시도해봐요)

- 저번주에 공부하고자했던 앱이 실행이되서 View가 그려지는 과정까지를 하지못했다.. 다음주라도 시간이 되면 하고싶다.


### 학습하는 데 조금이라도 도움이 된 사람이 있다면 칭찬해주세요!

- 모든 조원님들이 학습하시다가 좋은 자료가 있으면 공유해주셔서 너무 감사했습니다!

**Reply by Jinsujin** [reply]
멘탈...소중하죠...! 잘 쉬는것도 이후에 집중하는데 도움이 되요! 
이제 몸아프지 말기...⭐️

**Comment by bibi6666667** [comment]
### 학습하면서 잘했던 점, 좋았던 점은 무엇인가요? (다음주에도 계속 유지해보세요)
- 사진액자 미션을 포기하지 않고 월요일까지 끝낸 것
- 1일 1PR을 위해 노력한 것
- 피드백 내용을 최대한 반영하며 공부한 것

### 학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 - 그래서 어떻게 개선할 수 있을까요? (다음주에는 개선해봐요)
- 진도를 많이 못 나간 것ㅜㅜ
- 뭔가.. 공부에 해이해진 것. 초심을 되찾자🥲

### 새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? (다음주에 시도해봐요)
- 이번 미션에 단위 테스트 적용해보기 (이번 주말에 해보려고 합니다)

### 학습하는 데 조금이라도 도움이 된 사람이 있다면 칭찬해주세요!
- 무엇이든 물어보살 Rosa.. 감사합니다🙇‍♀️

**Reply by Jinsujin** [reply]
아이구 저야말로 비비와 함께 해서 고마웠어요...♥️ 
초심 찾으려면 게더오셔서 같이 라이브코딩 해보시는건 어떤가요....😊

**Comment by SongTaehwan** [comment]
### 학습하면서 잘했던 점, 좋았던 점은 무엇인가요? (다음주에도 계속 유지해보세요)
- 첫 주 미션을 마무리 한 점
- 코드를 작성하면서 최대한 많은 고민을 하려고 노력한 점

### 학습하면서 스스로 아쉬웠던 점, 부족했던 점은 무엇인가요 - 그래서 어떻게 개선할 수 있을까요? (다음주에는 개선해봐요)
- 진도를 많이 나가지 못한 점
- 이론 학습에 너무 많은 시간을 쏟은 것 같다.

### 새롭게 알게 되거나 시도하지 못했지만, 다음주에 도전해볼건 무엇인가요? (다음주에 시도해봐요)
- MVC, MVVM 패턴에 대해 간략하게 학습해보고 코드에 적용해 보기
- TDD 학습하기

### 학습하는 데 조금이라도 도움이 된 사람이 있다면 칭찬해주세요!
- 한 주 동안 학습자료 공유와 편하게 질문할 수 있는 분위기를 조성해준 2조 조원분들 모두 감사했습니다 👍
